WordPress/Classes/ViewRelated/Post/PostListViewController.swift

```diff
@@ -0,0 +1,685 @@
+        let blog = apost.blog
+        
+        if blog.supports(.Stats) {
+            // Needs Jetpack.
+            return
+        }
+        
+        WPAnalytics.track(.PostListStatsAction, withProperties: propertiesForAnalytics())
+        
+        // Push the Stats Post Details ViewController
+        let identifier = NSStringFromClass(StatsPostDetailsTableViewController.self)
+        let service = BlogService(managedObjectContext: ContextManager.sharedInstance().mainContext)
+        let statsBundle = NSBundle(forClass: WPStatsViewController.self)
+        let path = statsBundle.pathForResource("WordPressCom-Stats-iOS", ofType: "bundle")!
+        let bundle = NSBundle(path: path)
+        let statsStoryboard = UIStoryboard(name: self.dynamicType.statsStoryboardName, bundle: bundle)
+        let controller = statsStoryboard.instantiateViewControllerWithIdentifier(identifier) as! StatsPostDetailsTableViewController
+        
+        controller.postID = apost.postID
+        controller.postTitle = apost.titleForDisplay()
+        controller.statsService = WPStatsService(siteId: blog.dotComID, siteTimeZone: service.timeZoneForBlog(blog), oauth2Token: blog.authToken, andCacheExpirationInterval: self.dynamicType.statsCacheInterval)
+        
+        navigationController?.pushViewController(controller, animated: true)
+    }
+    
+    // MARK: - Filter Related
+    
+    override internal func currentPostAuthorFilter() -> PostAuthorFilter {
+        if !canFilterByAuthor() {
+            // No REST API, so we have to use XMLRPC and can't filter results by author.
+            return .Everyone
+        }
+        
+        if let filter = NSUserDefaults.standardUserDefaults().objectForKey(self.dynamicType.currentPostAuthorFilterKey) {
+            if filter.unsignedIntegerValue == PostAuthorFilter.Everyone.rawValue {
+                return .Everyone
+            }
+        }
+        
+        return .Mine
+    }
+
+    override internal func setCurrentPostAuthorFilter(filter: PostAuthorFilter) {
+        guard filter != currentPostAuthorFilter() else {
+            return
+        }
+        
+        WPAnalytics.track(.PostListAuthorFilterChanged, withProperties: propertiesForAnalytics())
+        
+        NSUserDefaults.standardUserDefaults().setObject(filter.rawValue, forKey: self.dynamicType.currentPostAuthorFilterKey)
+        NSUserDefaults.resetStandardUserDefaults()
+        
+        recentlyTrashedPostObjectIDs?.removeAllObjects()
+        resetTableViewContentOffset()
+        updateAndPerformFetchRequestRefreshingCachedRowHeights()
+        syncItemsWithUserInteraction(false)
+    }
+
+    private func keyForCurrentListStatusFilter() -> String {
+        return self.dynamicType.currentPostListStatusFilterKey
+    }
+
+    // MARK: - Cell Delegate Methods
+
+    func cell(cell: UITableViewCell!, receivedEditActionForProvider contentProvider: WPPostContentViewProvider!) {
+        let apost = contentProvider as! AbstractPost
```

Comment:
> However a crash is a worse experience

The whole idea behind making the app fail is to make sure we don't code to allow invalid internal states that should not be there in the first place.  It's the same principle as with assertions.  You try to make it fail as soon and loudly as possible so it comes out during testing, as opposed to the log, which is much easier to ignore (intentionally or not), and more likely to make it to release.